__tests__/app.test.js

```diff
@@ -114,12 +114,79 @@ describe("GET /api/articles", () => {
       expect(articles).toBeSortedBy('created_at', { descending: true });
     });
   });
-  test("400: Responds with Bad Request: Invalid Query when passed an invalid sort_by or order_by query ", () => {
+  // test("400: Responds with Bad Request: Invalid Query when passed an invalid sort_by or order_by query ", () => {
+  //   return request(app)
+  //   .get("/api/articles?sort_by=name")
+  //   .expect(400)
+  //   .then(({ body })=>{
+  //     expect(body.message).toBe('Bad Request: Invalid Query')
+  //   });
+  // });
+});
+
+describe("GET /api/articles/:article_id/comments", () => {
+  test("200: Responds with an array of comments for the given article_id, sorted by most recent first", () => {
     return request(app)
-    .get("/api/articles?sort_by=name")
-    .expect(400)
-    .then(({ body })=>{
-      expect(body.message).toBe('Bad Request: Invalid Query')
-    });
+      .get("/api/articles/9/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const { comments }  = body;
+        expect(comments).toHaveLength(2)
+        comments.forEach((comment) => {
+          expect(comment).toHaveProperty("comment_id");
+          expect(comment).toHaveProperty("votes");
+          expect(comment).toHaveProperty("created_at");
+          expect(comment).toHaveProperty("author");
+          expect(comment).toHaveProperty("body");
+          expect(comment).toHaveProperty("article_id");
+        });
+        expect(comments[0]).toMatchObject({
+          body: "Oh, I've got compassion running out of my nose, pal! I'm the Sultan of Sentiment!",
+          votes: 16,
+          author: "butter_bridge",
+          article_id: 9,
+        });
+        expect(comments[1]).toMatchObject({
+          body: "The owls are not what they seem.",
+          votes: 20,
+          author: "icellusedkars",
+          article_id: 9,
+        });
+        expect(typeof comments[0].created_at).toBe("string");
+      });
   });
-});
+  test("200: Responds with an array sorted by created_at in descending order", () => {
+    return request(app)
+      .get("/api/articles/9/comments")
+      .expect(200)
+      .then(({ body })=>{
+        const { comments } = body;
+        expect(comments).toBeSortedBy('created_at', { descending: true });
+      });
+  });
+  test("404: Responds with Not Found when passed a non-existent article_id", () => {
+    return request(app)
+      .get("/api/articles/864/comments")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.message).toBe("Not Found: no article found for article_id 864");
```

Comment:
Nice, if you leave this message as "Not found" then you should be able to use that same error to handle other errors, not just comments.